baybe/utils/plotting.py

```diff
@@ -0,0 +1,126 @@
+"""Plotting utilities."""
+
+import json
+import os
+import sys
+import warnings
+from pathlib import Path
+
+from matplotlib.axes import Axes
+
+
+def create_example_plots(
+    figure: Axes,
+    path: Path,
+    base_name: str,
+) -> None:
+    """Create plots from a figure and save them as a svg file.
+
+    The plots will be saved in the location specified by ``path``.
+    The attribute ``base_name`` is used to define the name of the outputs.
+
+    If the ``SMOKE_TEST`` variable is set, no plots are being created and this method
+    immediately returns.
+
+    The function attempts to read the predefined themes from ``plotting_themes.json``.
+    For each theme it finds, a file ``{base_name}_{theme}.svg`` is being created.
+    If the file cannot be found, if the JSON cannot be loaded or if the JSON is not well
+    configured, a fallback theme is used.
+
+    Args:
+        figure: The Axes object containing the figure that should be plotted.
+        path: The path to the directory in which the plots should be saved.
+        base_name: The base name that is used for naming the output files.
+    """
+    # Check whether we immediately return due to just running a SMOKE_TEST
+    if "SMOKE_TEST" in os.environ:
+        return
+
+    # Lazily import optional packages which are now required
+    import matplotlib.pyplot as plt
```

Comment:
I'm not quite sure what you two discussed about the optional dependencies here (I don't actually think a delayed import is necessary because the function will not be called anyway unless the examples are executed, in which case matplotlib is available – which is different from, say, the substance parameter, which can be imported even though chem deps are not installed). I'll leave that for you two to figure out. However, I can clearly say what does NOT make sense is that there is BOTH a lazy import of matplotlib inside the function AND a top level import

Reply:
That makes sense. Will move this to the top again, we can still change this once we do the major example overhaul